urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/driver/Phase1.java

```diff
@@ -0,0 +1,244 @@
+        verifyMathematicalOperations();
+        verifyBooleanChecks();
+        verifyComparisons();
+        verifyFlyweight();
+    }
+
+    private static void makeSomeTestPriceObjects() throws Exception {
+        System.out.println("Creating some Test Price Objects.");
+        testPriceHolder.add(PriceFactory.makeLimitPrice("10.50"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("$1400.99"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("$-51.52"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice(".49"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("-0.89"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("12"));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("90."));
+        testPriceHolder.add(PriceFactory.makeLimitPrice("14.5"));
+        testPriceHolder.add(PriceFactory.makeMarketPrice());
+    }
+
+    private static void verifyTestPriceValues() {
+        System.out.println("Verifying the Values in Your Test Price Objects:");
+        String format = "%-9s --> %9s : %s%n";
+        System.out.format(format, "$10.50", testPriceHolder.get(0), testPriceHolder.get(0).toString().equals("$10.50") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$1,400.99", testPriceHolder.get(1), testPriceHolder.get(1).toString().equals("$1,400.99") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$-51.52", testPriceHolder.get(2), testPriceHolder.get(2).toString().equals("$-51.52") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$0.49", testPriceHolder.get(3), testPriceHolder.get(3).toString().equals("$0.49") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$-0.89", testPriceHolder.get(4), testPriceHolder.get(4).toString().equals("$-0.89") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$12.00", testPriceHolder.get(5), testPriceHolder.get(5).toString().equals("$12.00") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$90.00", testPriceHolder.get(6), testPriceHolder.get(6).toString().equals("$90.00") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$14.50", testPriceHolder.get(7), testPriceHolder.get(7).toString().equals("$14.50") ? "CORRECT" : "ERROR");
+        System.out.format(format, "MKT", testPriceHolder.get(8), testPriceHolder.get(8).toString().equals("MKT") ? "CORRECT" : "ERROR");
+        System.out.println();
+    }
+
+    private static void verifyMathematicalOperations() {
+        System.out.println("Verifying the Functionality of your Mathematical Operations:");
+        String format = "%-9s %c %9s = %9s : %s%n";
+        try {
+            Price results = testPriceHolder.get(0).add(testPriceHolder.get(1));
+            System.out.format(format, testPriceHolder.get(0), '+', testPriceHolder.get(1), results, results.toString().equals("$1,411.49") ? "CORRECT" : "ERROR");
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("FAILED: " + ex.getMessage());
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 59:64, 65:70, 71:76, 83:88, 89:94.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.